refactor flask app

ttung opened the pull request:
### Description
Break up app.py into the flask app initialization and the routes.  In the long term, we will want to split up the routes into separate python modules and this simplifies that future step.

Depends on #107 

### Test plan
1. pytest
1. run locally
1. run on eb

phoenixAja commented:
oh no you beat me, i wanted to do this after this demo deadline.

phoenixAja commented on src/py/aspen/flask_app/flask_app.py:
```diff
@@ -0,0 +1,47 @@
+import os
```
can we name this src/py/aspen/app/app ?

ttung replied:
probably.  we'd need to change our entry point as well.

phoenixAja replied:
i just think the flask part is unnecessary, but if it's too annoying i don't mind waiting until later

ttung replied:
no problem.

phoenixAja commented on src/py/aspen/flask_app/routes.py:
```diff
@@ -0,0 +1,72 @@
+from pathlib import Path
```
can we rename this src/py/aspen/app/views ?

phoenixAja reviewed (approved):
this looks good to me, just wanted to rename some stuff

ttung commented:
ok it's all been renamed.

Commit:
refactor flask app

### Description
Break up app.py into the flask app initialization and the routes.  In the long term, we will want to split up the routes into separate python modules and this simplifies that future step.

### Test plan
1. pytest
1. run locally
1. run on eb
